### PR TITLE
Improvements on paywall and open graph image

### PR DESCRIPTION
**Pull request by felpsio**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "scripts": {
         "start": "gatsby develop",
         "clean": "gatsby clean",
-        "build": "gatsby build",
+        "build": "gatsby clean && gatsby build",
```
Comment (Swizec) [inline package.json]: A better approach is using `npm run clean` so you ensure running `yarn clean` and `yarn build` use the same steps. Defaulting to `npm run` in package.json means you aren't sneakily requiring yarn.

File: src/components/layout.js [hunk 2]
```diff
@@ -14,33 +14,41 @@ import QuickThanks from "./quickthanks"
...
-          if (child.id === 'lock') isLocked = true;
-          if (isLocked === true ) {
-            child.style.display = 'none';
+
+  useEffect(() => {
+    if (
+      !window.localStorage.getItem("unlock_handbook") ||
+      !window.localStorage.getItem("sale_id")
+    ) {
+      setTimeout(
```
Comment (Swizec) [inline src/components/layout.js]: window.requestAnimationFrame is better here I think.

How come you can't do this in the same hook as the other paywall stuff?

File: src/components/layout.js [hunk 3]
```diff
@@ -14,33 +14,41 @@ import QuickThanks from "./quickthanks"
...
+  }, [page])
+
+  useLayoutEffect(() => {
+    if (typeof window !== "undefined" ) {
 
       window.requestAnimationFrame(() => {
-        const overlay = document.createElement("div")
-        const main = document.querySelector("main#content")
+        const overlay = typeof window !== 'undefined' && document.createElement("div")
+        const main    = typeof window !== 'undefined' && document.querySelector("main#content")
```
Comment (Swizec) [inline src/components/layout.js]: You've already verified `window` exists by now so you don't have to check again.

File: src/components/layout.js [hunk 4]
```diff
@@ -116,12 +126,12 @@ const Sidebar = props => {
...
+        {
+          showPaywall === false
+            && <Reactions />
+        }
         {showPaywall ? (
-          <>
-            <Paywall floating />
-            <Paywall />
-          </>
+          <Paywall page={props.location.pathname} />
```
Comment (Swizec) [inline src/components/layout.js]: Why is it useful for the paywall to get the page name?

File: src/components/logo.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import React from "react"
 import styled from "@emotion/styled"
-import { keyframes } from "@emotion/core"
-import { useThemeUI } from "theme-ui"
+// import { keyframes } from "@emotion/core"
+// import { useThemeUI } from "theme-ui"
```
Comment (Swizec) [inline src/components/logo.js]: You can remove this

File: src/components/logo.js [hunk 6]
```diff
@@ -11,72 +11,72 @@ const Svg = styled(({ width, height, ...props }) => (
...
+// `
 
-const ElectronStatic2 = styled("circle")`
-  transform-origin: 50% 50%;
-  transform: rotate3d(2, 0, 1, 190deg);
-`
+// const ElectronStatic2 = styled("circle")`
+//   transform-origin: 50% 50%;
+//   transform: rotate3d(2, 0, 1, 190deg);
+// `
```
Comment (Swizec) [inline src/components/logo.js]: You can remove all this

File: src/components/logo.js [hunk 7]
```diff
@@ -11,72 +11,72 @@ const Svg = styled(({ width, height, ...props }) => (
...
+  //   <g>
+  //     <ElectronStatic1 {...electronProps} stroke="#f90" />
+  //     <ElectronStatic2 {...electronProps} stroke="magenta" />
+  //   </g>
+  // ) : (
+  //   <g>
+  //     <Electron1 {...electronProps} />
+  //     <Electron2 {...electronProps} />
+  //   </g>
+  // )
```
Comment (Swizec) [inline src/components/logo.js]: This can go

File: src/pages/serverless-flavors.mdx [hunk 8]
```diff
@@ -66,6 +66,8 @@ Not sure why that is.
 
 [![](../images/firebase.png)](https://firebase.google.com/)
 
+<div id="firebase" />
```
Comment (Swizec) [inline src/pages/serverless-flavors.mdx]: Shouldn't this say `<div id="lock" />` ?

File: src/components/header.js [hunk 9]
```diff
@@ -49,12 +49,36 @@ const Dot = props => (
...
 
+  const updateLockedContent = () =>  {
+
+    if (
+      !window.localStorage.getItem("unlock_handbook") ||
+      !window.localStorage.getItem("sale_id")
+    ) {
+      let children = document.getElementById('content').children //[].slice.call(document.getElementById('content').children);
+
+      setTimeout(
```
[the remark this answers is not in the document]
Comment (felpsio) [reply src/components/header.js]: I duplicated. Change the colors also change the children and I didn't find a way to get the trigger for this in layout

File: src/components/layout.js (hunk 4 above)
Comment (felpsio) [reply src/components/layout.js]: To get the changes when the user changes the page. According to the way react update the children the `lock content function` wasn't fired if the user just change the page

**Pull request merged**